src/poetry/console/commands/group_command.py

```diff
@@ -107,3 +111,36 @@ def project_with_activated_groups_only(self) -> ProjectPackage:
         return self.poetry.package.with_dependency_groups(
             list(self.activated_groups), only=True
         )
+
+    def validate_group_options(self, group_options: dict[str, set[str]]) -> bool:
+        """
+        Currently raises en error if it detects that a group is
+        not part of pyproject.toml
+
+        Can be overridden to adapt behavior.
+        """
+        invalid_options = defaultdict(set)
+        for opt, groups in group_options.items():
+            for group in groups:
+                if not self.poetry.package.has_dependency_group(group):
+                    invalid_options[opt].add(group)
+        if invalid_options:
+            line_err = (
+                "<error>The <fg=yellow;options=bold>--with</>, "
+                "<fg=yellow;options=bold>--without</>, "
+                "and <fg=yellow;options=bold>--only</> "
+                "options may only have valid groups."
+            )
+            for opt, invalid_groups in invalid_options.items():
+                line_err += (
+                    " <fg=red;options=bold>Invalid"
+                    f" [{', '.join(sorted(invalid_groups))}] provided to --{opt}.</>"
+                )
+            line_err += "</error>"
+            self.line_error(line_err)
+        if len(invalid_options) != 0:
```

Comment:
```suggestion
        if invalid_options:
```

Reply:
Fixed in d9aaafe64f8602e6832a2f098275dc9ed0ff7c56